### PR TITLE
style: apply new designs to field configuration (x, y)

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: https://github.com/lightdash/lightdash/issues/11470

### Description:

See main ticket and request figma link

Screenshot 2024-10-23 at 16 34 56

THis PR: 
- Introduces `usePillSelectStyles` - allows us to create PillSelect's with Mantine hooks
- Allows the user to update the sort and aggregation inside the fieldReferenceSelect
- Replaces the `+ add ` button with an action icon
- Replaces the `clearable` prop with a different approach (with the - action icon)
- Refactors the `setSortBy` action. 

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by joaoviana** [comment]
If this meets the acceptable requirements @PriPatel , could we have @magnew review this please? 

So essentially after your design review -> we can get Marshall to review if the remaining bits can be done separately? What do you think?

**Comment by PriPatel** [comment]
Review here! 
https://www.notion.so/lightdash/Review-128a63207a7a80bc8ea8c4f061bd865a

- There is one big logic change that needs to take a place and a few styling bits that need to match Figma (less important than the logic change which might break charts if we proceed this way!)

File: packages/frontend/src/components/DataViz/store/cartesianChartBaseSlice.ts [hunk 1]
```diff
@@ -284,10 +284,23 @@ export const cartesianChartConfigSlice = createSlice({
...
+            if (action.payload.direction) {
+                fieldConfig.sortBy = [
+                    {
+                        reference: action.payload.reference,
+                        direction: action.payload.direction,
+                    },
+                ];
+            } else {
+                fieldConfig.sortBy = undefined;
+            }
```
Comment (joaoviana) [inline packages/frontend/src/components/DataViz/store/cartesianChartBaseSlice.ts:303]: can clear now the sortBy

File: packages/frontend/src/components/DataViz/hooks/usePillSelectStyles.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import { createStyles } from '@mantine/core';
+
+export const usePillSelectStyles = createStyles(
```
Comment (joaoviana) [inline packages/frontend/src/components/DataViz/hooks/usePillSelectStyles.tsx:3]: this helps with creating a pill select
image

**Comment by PriPatel** [comment]
@joaoviana Just remembered(!), as we're not doing the full milestone, the tooltip should be added here! 'Aggregation type' when you hover over the y pill and 'Sort by' on the x axis pill

**Comment by joaoviana** [comment]
@PriPatel added tooltips!

@magnew  - updated e2e tests and added test ids where necessary

File: packages/frontend/src/components/DataViz/config/DataVizAggregationConfig.tsx [hunk 3]
```diff
@@ -72,40 +80,46 @@ export const DataVizAggregationConfig: FC<Props> = ({
...
+                    value && onChangeAggregation(value)
+                }
+                classNames={{
+                    item: classes.item,
+                    dropdown: classes.dropdown,
+                    input: classes.input,
+                    rightSection: classes.rightSection,
+                }}
+                styles={{
+                    input: {
```
Comment (magnew) [inline packages/frontend/src/components/DataViz/config/DataVizAggregationConfig.tsx:115]: I can't remember: is the classNames styling no good when a parameter is required (aggregation here)? It's a little odd that we are stying input with styles and classnames. But maybe that's life in modern react 😂 . I'm sure this isn't the only place we are doing things like that.

File: packages/frontend/src/components/DataViz/hooks/usePillSelectStyles.tsx (hunk 2 above)
Comment (magnew) [reply packages/frontend/src/components/DataViz/hooks/usePillSelectStyles.tsx:3]: Cool!

**Review by magnew — APPROVED** [review]
✅ Looks great
✅ Works great - I don't see any issues
✅ Code looks good!

**Pull request merged**